### PR TITLE
router: improve packaging and remove inherits

### PR DESCRIPTION
**Pull request by pablopalacios**
`fluxible-router` packaging works in the same way as `fluxible-addons-react` now. Summary of changes:

- converted cjs modules into es modules;
- added babel runtime as dependency;
- `dist` now builds cjs and an es package;
- merged addons and lib folders since package is now tree-shakeable;
- removed inherits in favor of standard es classes;
- updated docs;

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by redonkulus** [comment]
Thanks for this. One thing to keep in mind is that we had already start some refactoring of this library in this branch https://github.com/yahoo/fluxible/tree/fluxible-router-1.x

The current 1.x npm alpha had this change in it. However, we have done releases from this branch over that one. I forget, but did you incorporate the changes from that branch in this one or did we lose it?

File: packages/fluxible-router/docs/api/HistoryWithHash.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ By default, the hash fragments are just url paths.  With `HistoryWithHash`, you
 
 ```js
 var handleHistory = require('fluxible-router').handleHistory;
```
Comment (redonkulus) [inline packages/fluxible-router/docs/api/HistoryWithHash.md:29]: should we change these to import?

File: packages/fluxible-router/package.json [hunk 2]
```diff
@@ -24,10 +27,10 @@
     }
   ],
   "dependencies": {
+    "@babel/runtime": "^7.14.0",
```
Comment (redonkulus) [inline packages/fluxible-router/package.json:30]: Should this be a devDependency at the repo root?

**Comment by pablopalacios** [comment]
> Thanks for this. One thing to keep in mind is that we had already start some refactoring of this library in this branch https://github.com/yahoo/fluxible/tree/fluxible-router-1.x
> 
> The current 1.x npm alpha had this change in it. However, we have done releases from this branch over that one. I forget, but did you incorporate the changes from that branch in this one or did we lose it?

Yeah, it's a little bit messy now. But I've checked again and all the changes from that branch were applied here ;)

File: packages/fluxible-router/package.json (hunk 2 above)
Comment (pablopalacios) [reply packages/fluxible-router/package.json:30]: @babel/runtime is a runtime dependency so it can't be devDependency. And we need to have on each package.json since otherwise it would not be installed when one do `npm install fluxible-router`.

File: packages/fluxible-router/docs/api/HistoryWithHash.md (hunk 1 above)
Comment (pablopalacios) [reply packages/fluxible-router/docs/api/HistoryWithHash.md]: I've updated the examples to use modern es ;)

File: packages/fluxible-router/package.json (hunk 2 above)
Comment (redonkulus) [reply packages/fluxible-router/package.json:30]: whats the reason for needing it to be a runtime dependency?

**Comment by redonkulus** [comment]
Ok that other branch will just be removed once we merge this branch into master. Thanks for keeping things in sync.

File: packages/fluxible-router/docs/api/README.md [hunk 3]
```diff
@@ -1,29 +1,21 @@
...
+  handleHistory,
+  handleRoute,
+  navigateAction,
+  NavLink,
+  RouteStore,
+  HistoryWithHash
+} from 'fluxible-router';
 ```
-
-## Addons
```
Comment (redonkulus) [inline packages/fluxible-router/docs/api/README.md:17]: why was this removed?

Comment (pablopalacios) [reply packages/fluxible-router/docs/api/README.md:17]: Since `HistoryWithHash` is part of the main package now, I kept the same doc structure as for the other functions/classes. So an entry was added on line 8 for it.

Comment (pablopalacios) [reply packages/fluxible-router/docs/api/README.md:17]: By the way, since the package is now tree-shakeable, there is no need to keep the addons in a separated folder. Also, when `fluxible-router` was changed to use babel, the import for this package got broken:

- v1: `import { HistoryWithHash } from 'fluxible-router/addons'`
- v2: `import { HistoryWithHash } from 'fluxible-router/dist/addons'`
- This PR:  `import { HistoryWithHash } from 'fluxible-router'`

File: packages/fluxible-router/package.json (hunk 2 above)
Comment (pablopalacios) [reply packages/fluxible-router/package.json:30]: Oh, it's the same as we did for `fluxible-addons-react`. I explained it here: https://github.com/yahoo/fluxible/pull/683#issuecomment-830857300

File: packages/fluxible-router/package.json [hunk 4]
```diff
@@ -2,13 +2,16 @@
   "name": "fluxible-router",
   "version": "2.0.0-beta.12.0",
   "description": "Routing for Fluxible applications",
-  "main": "index.js",
+  "main": "./dist/lib/index.js",
```
Comment (redonkulus) [inline packages/fluxible-router/package.json:5]: I think you can just remove the dist folder and just having everything off of "lib" since its all under "src" now.

File: packages/fluxible-router/package.json (hunk 2 above)
Comment (redonkulus) [reply packages/fluxible-router/package.json:30]: oh right, totally forgot your great explanation. do we need to track the hardcoded babel version to remove that later? any follow ups with this approach we need to keep an eye on later?

File: packages/fluxible-router/package.json (hunk 4 above)
Comment (pablopalacios) [reply packages/fluxible-router/package.json:5]: Oh, my fault, this lib is not the same as the previous lib. I'll rename it to cjs to make it clear.

Comment (pablopalacios) [reply packages/fluxible-router/package.json]: Done.

File: packages/fluxible-router/package.json [hunk 5]
```diff
@@ -2,13 +2,16 @@
   "name": "fluxible-router",
   "version": "2.0.0-beta.12.0",
   "description": "Routing for Fluxible applications",
-  "main": "index.js",
+  "main": "./dist/cjs/index.js",
```
Comment (redonkulus) [inline packages/fluxible-router/package.json:5]: Do you need a module property for the es modules code now too?

Comment (pablopalacios) [reply packages/fluxible-router/package.json:5]: Yes! You are completely right!

File: packages/fluxible-router/package.json (hunk 2 above)
Comment (pablopalacios) [reply packages/fluxible-router/package.json:31]: We could do as react-redux: https://github.com/reduxjs/react-redux/blob/master/.babelrc.js#L16. They read the version from package.json removing the possibility to have mismatched versions between the installed one and the consumed one. 

It's nice, but, in my opinion, giving that we don't update this package too often, we could keep it simple (as we did on addons-react).

Regarding future, it's nice to keep the package version updated since we can benefit more from newer utilities or see them completely vanish when browsers evolve. But the same is valid for all the other babel packages. So I would say that the only thing to keep in mind is, whenever we use update babel deps, we should update this one as well (and remember to update babelrc to point to the right version).

But if prefer, I can prepare another commit that reads the version from package.json as well. What do you say?

Comment (redonkulus) [reply packages/fluxible-router/package.json:31]: I sort of like the automatic version update that react-redux does. Its one less thing to remember when upgrading babel, because the version will get stale as new versions come out. I think you could do that in a follow up PR for both packages.

**Review by redonkulus — APPROVED** [review]

**Comment by redonkulus** [comment]
Released in fluxible-router@2.0.0-beta.13.0
